### PR TITLE
33: Eclipse Naming conventions, 34: Correct license headers

### PR DESCRIPTION
**Pull request by keilw**
Task-Url: https://github.com/microprofile/microprofile-samples/issues/33,
https://github.com/microprofile/microprofile-samples/issues/issues/34

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!--
```
Comment (kenfinnigan) [inline pom.xml]: Why the change from 4 to 8 spaces?

File: pom.xml [hunk 2]
```diff
@@ -18,528 +18,571 @@
...
-                                    </transformers>
-                                </configuration>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
+					<plugin>
```
Comment (kenfinnigan) [inline pom.xml]: A few things with this:

- Why part of the "skipTests" profile? If "skipTests" isn't active it doesn't run.
- If I'm not mistaken, wouldn't this check for an author of "Antonio Goncalves and others" in headers? What about new files that are added by anyone other than Antonio? I believe they should have the authorship set to themselves, which would cause this plugin to fail the check.

Comment (keilw) [inline pom.xml]: Rat does not update headers, but for now I decided to extend and leverage the profile for "license checks". 
Antonio was the initial author for all files so similar to e.g. the test module in the config project there is a "leading" or initial author (Mark S in its case)
We have the same situation if say Emily/IBM or someone else added something to the part Mark started. That's what the "others" are for. If you look at JBoss headers (I trust you know some ;-) it says "others by the @author tag", so anybody who contributes to existing files would also get credit this way. Guess that's how individual contributors get credit in a JBoss project, so it's the same here. If more samples have multiple leads, the license checker can also be split across different modules. It just did not seem necessary at first.

File: pom.xml (hunk 1 above)
Comment (keilw) [inline pom.xml]: Do you mean just POMs or all files? Maybe it was tabs instead of spaces? Only the POMs are manually edited and (for the new plugins) also formatted. Everything else if something was reformatted was done by the license checker plugin.

File: pom.xml (hunk 2 above)
Comment (kenfinnigan) [inline pom.xml]: My point @keilw was that an entirely new file added by someone other than Antonio would not have "Antonio Goncalves and others" it would have "IBM and others" or even "IBM" to start with.

Original authorship is important here. Although Antonio was the original author of the current files, an entirely new file in the repository from someone other than Antonio would not have "Antonio Goncalves and others" as the author.

**Comment by keilw** [comment]
The built-in Eclipse/Java formatting rules e.g. have a tab size 8. In a different thread and repo we discussed another plugin that applies Eclipse formatting rules in Maven (similar to https://github.com/unitsofmeasurement/unit-ri/blob/master/src/etc/eclipse-formatter-config.xml) but let's keep that in a separate PR and issues. Is there a problem with the way I applied Eclipse package and correct headers?

File: pom.xml (hunk 1 above)
Comment (kwsutter) [inline pom.xml]: I agree with Ken.  I don't care for the further indentation, especially in the POM files...  Some of the stanzas are indented so far, it makes for difficult reading.  Can the plugin be adjusted to only do 4 spaces per "tab" instead of 8?  And, are the tabs actually converted to spaces instead of tab characters (I hope so).  Thanks.

File: pom.xml (hunk 2 above)
Comment (kwsutter) [inline pom.xml]: @kenfinnigan, I'm missing where the automatic check is being done for "Antonio Goncalves...", but if that's the case, then I agree we shouldn't have that in the automated check.  Whether we're talking about new files or even updates to the current ones, we don't want to be tied to the current Copyright.

Comment (kenfinnigan) [inline pom.xml]: @kwsutter My understanding is that https://github.com/microprofile/microprofile-samples/pull/36/files/dbf2160fb91214779030e2569218d6f9ecdae15e#diff-600376dffeb79835ede4a0b285078036R310 will provide "Antonio Goncalves" as a property to the license check plugin, which is present in an activeByDefault profile.

The check would then fail if anything other than "Antonio Goncalves" is present.

File: pom.xml (hunk 1 above)
Comment (keilw) [inline pom.xml]: There is no Formatter plugin (except if your IDE automatically formats a file like aPOM) right now. At most the license plugin could have applied some default, but I don't think it does. Happy to apply 
<groupId>net.revelc.code</groupId>
				<artifactId>formatter-maven-plugin</artifactId> https://github.com
we used in projects like https://github.com/unitsofmeasurement/unit-ri/blob/master/pom.xml (actually based on suggestions by LJC in the JCP EC) and transfer it here by updating the fork and PR.

Comment (kwsutter) [inline pom.xml]: Looks much better, @keilw.  Thanks.

File: pom.xml (hunk 2 above)
Comment (kwsutter) [inline pom.xml]: Thanks, @kenfinnigan.  I missed that configuration for the plugin.  @keilw, can we get this resolved?  Once we clean up this plugin configuration, I think we can merge in these changes to get this repo cleaned up for the move.

**Comment by keilw** [comment]
@kwsutter @kenfinnigan  You mean the shade-plugin in the "hammock" profile? I did not run or change that particular profile, maybe @johnament knows more about it, or offer PR only on that detail

**Comment by keilw** [comment]
If it's only the POM (those did not get automatic headers via the plugin) I'll fix it manually.
